onnxruntime/core/framework/execution_frame.cc

```diff
@@ -353,8 +352,8 @@ Status ExecutionFrame::AllocateAsPerAllocationPlan(OrtValue& ort_value, int ort_
   ORT_ENFORCE(ort_value_index >= 0 && static_cast<size_t>(ort_value_index) < alloc_plan.size());
   const auto& per_alloc_plan = alloc_plan[ort_value_index];
 
-  auto alloc_info = per_alloc_plan.location;
-  auto ml_type = per_alloc_plan.value_type;
+  const auto& alloc_info = per_alloc_plan.location;
```

Comment:
We should probably also make the copy constructor private to prevent future such cases.

Reply:
I first made the assignment operator private but there are legitimate uses for that. Setting the location in the allocation plan in the first place, and in Tensor::operator=

This is a minor thing but a bit over 0.1% of perf for one model I tested.